### DISCUSSION TITLE
Texture intensity over tilt angle plot

### DISCUSSION THREAD

**Discussion by 90Maria** [mtex-toolbox/mtex#696] [Ask Anything] [answered]
Hello,

I would like to plot texture intensity over tilt angle in RD, TD and preferably 45° for 0001 pole figure. I know this should be possible. Unfortunately my MTEX-skills are not good enough and maybe someone would be so kind and provide me his script ?

Thanks a lot 
Maria

**Comment by lfgmorales** [comment]
Hi Maria


Could you expand a little bit more about what do you want to do? Do you want to rotate to your data to a different position from that used in the measurements? The texture index (how strong is your texture) is not affected by rotations as far as I am aware of, but maybe you meant something different


for texture index calculation, have a look here


https://mtex-toolbox.github.io/ODF.textureindex.html


for data rotation, have a look here


https://mtex-toolbox.github.io/EBSD.rotate.html


cheers


L


Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________

**Reply by 90Maria** [reply]
Hey Luiz,

Thank you for your fast reply :)

I measured my textures with XRD. Either I would like to export the calculated texture intensity e.g. from the 0001 pole figure to create a plot that shows the texture intensity over the tilt angle.  And for the tilt angle I would like to have the transverse direction, longitudinal direction and 45° displayed in different plots. Or if that is easier, then directly in MTEX create different plots that show the texture intensity over the different tilt angles.

So actually nothing should be rotated at all. I would like to have a table where the calculated texture intensity and the different tilt angles are displayed. 


For example something like this graphic i would like to create:
![grafik](https://user-images.githubusercontent.com/77343995/104419948-78bd2e80-5579-11eb-82f2-77804a2644b6.png)

Greetings
Maria

**Comment by lfgmorales** [comment]
Hi Maria


I guess what you are looking for are essentially fibres of oirentatons?


Have a look here


https://mtex-toolbox.github.io/OrientationFibre.html


Have a look in "defining a fibre by direction"


Also you may want to look here


https://mtex-toolbox.github.io/ODFPoleFigure.html



cheers



Luiz





Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________

**Comment by kilir** [comment, marked as answer]
```
% some example data
pf = calcDensity(vector3d(0,0,1),'halfwidth',20*degree,'antipodal')
plot(pf)
% define the vectors along which you want to get the data
v = vector3d.byPolar(linspace(0,pi/2,30),pi/2,'antipodal');
hold on
plot(v,'MarkerFaceAlpha',0.2)
hold off
nextAxis
plot(v.theta/degree,eval(pf,v))
```

<img width="705" alt="intensity-along_some_points" src="https://user-images.githubusercontent.com/11697557/104428223-84622280-5584-11eb-9cc8-abd4988c3db1.png">

Cheers,
Rüdiger

**Reply by 90Maria** [reply]
First of all, thank you for the great help...
It almost works now, but I can't get the graph to show up, it always gives me this error.

```
Error using eval
Must be a string scalar or character vector.

Error in neu (line 112)
plot(v.theta/degree,eval(pf1,v))
```

Any idea how I can change the data typ (variable)?



```
%% test

pf1 = calcPoleFigure(odf_cor,[Miller(0,0,0,2,CS)],'halfwidth',20*degree,'antipodal');

figure;

plot(pf1)
% define the vectors along which you want to get the data
v = vector3d.byPolar(linspace(0,pi/2,30),pi/2,'antipodal');
hold on
plot(v,'MarkerFaceAlpha',0.2)
hold off
plot(v.theta/degree,eval(pf1,v))
```

Cheers,
Maria

**Reply by kilir** [reply]
Hi,
`calcPoleFigure` will derive discrete values on a grid (actually which you could also read out and plot those for your application) while `calcPDF` computes a continuous function. So if you already have an ODF, you should be fine with `calcPDF`.
```
pf1 = calcPDF(odf,Miller(0,0,0,2,odf.CS));
plot(pf1)
% define the vectors along which you want to get the data
v = vector3d.byPolar(linspace(0,pi/2,30),pi/2,'antipodal');
hold on
plot(v,'MarkerFaceAlpha',0.2)
hold off
nextAxis
plot(v.theta/degree,eval(pf1,v))
```

Cheers,
Rüdiger

**Reply by 90Maria** [reply]
Thank you very much for the help. The script works perfectly now :)

Grüße,
Maria
